hack/netpol/pkg/main/main.go

```diff
@@ -0,0 +1,679 @@
+package main
+
+import (
+	"fmt"
+	. "github.com/vmware-tanzu/antrea/hack/netpol/pkg/utils"
+	log "github.com/sirupsen/logrus"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type NetPolConfig struct {
+	pods       []string
+	namespaces []string
+	k8s        *Kubernetes
+}
+
+// common for all tests.  these get hardcoded into the Expect() clauses,
+// so, we cant easily parameterize them (well, we could, but that would
+// make the code harder to interpret.
+var pods []string
+var namespaces []string
+var p80 int
+var p81 int
+var allPods []Pod
+
+func init() {
+	p80 = 80
+	p81 = 81
+	pods = []string{"a", "b", "c"}
+	namespaces = []string{"x", "y", "z"}
+
+	for _, podName := range pods {
+		for _, ns := range namespaces {
+			allPods = append(allPods, NewPod(ns, podName))
+		}
+	}
+}
+
+func bootstrap(k8s *Kubernetes) {
+	k8s.CleanNetworkPolicies([]string{"x","y","z"})
+	//p81 := 81
+	for _, ns := range namespaces {
+		k8s.CreateOrUpdateNamespace(ns, map[string]string{"ns": ns})
+		for _, pod := range pods {
+			fmt.Println(ns)
+			k8s.CreateOrUpdateDeployment(ns, ns+pod, 1,
+				map[string]string{
+					"pod": pod,
+				})
+		}
+	}
+	k8s.CleanNetworkPolicies([]string{"x","y","z"})
+}
+
+func validate(k8s *Kubernetes, reachability *Reachability, port int) {
+	// better as metrics, obviously, this is only for POC.
+	for _, pod1 := range allPods {
+		for _, pod2 := range allPods {
+			log.Infof("Probing: %s, %s", string(pod1), string(pod2))
+			connected, err := k8s.Probe(pod1.Namespace(), pod1.PodName(), pod2.Namespace(), pod2.PodName(), port)
+			log.Infof("... expected %v , got %v", reachability.Expected.Get(string(pod1), string(pod2)), connected)
+			if err != nil {
+				log.Errorf("unable to make main observation on %s -> %s: %s", string(pod1), string(pod2), err)
+			}
+			reachability.Observe(pod1, pod2, connected)
+			if !connected {
+				if reachability.Expected.Get(string(pod1), string(pod2)) {
+					log.Warnf("FAILED CONNECTION FOR WHITELISTED PODS %s -> %s !!!! ", string(pod1), string(pod2))
+				}
+			}
+		}
+	}
+}
+
+func main() {
+	k8s, err := NewKubernetes()
+	if err != nil {
+		panic(err)
+	}
+	k8s.CleanNetworkPolicies(namespaces)
+
+	bootstrap(k8s)
+	testWrapperPort80(k8s, TestDefaultDeny)
```

Comment:
I am not sure what is going on here: you call `bootstrap(k8s)` both before calling `testWrapperPort80` and inside `testWrapperPort80`?

Reply:
^+1

Reply:
yeah some redundant calls. ill clean them up good catch